### PR TITLE
JBDS-4482 Create 'About' dialog to show most relevant information about installer

### PR DESCRIPTION
**Pull request by sudhirverma**

### CODE REVIEW

File: browser/app.css [hunk 1]
```diff
@@ -21,7 +21,7 @@ html {
   z-index: 6;
 }
 
-footer {
+.footer {
```
Comment (dgolovin) [inline browser/app.css]: Changing footer class breaks confirmation page layout

image

File: main/index.js [hunk 2]
```diff
@@ -102,3 +106,26 @@ app.on('ready', function() {
     }
   });
 });
+
+function openAboutWindow() {
```
Comment (dgolovin) [inline main/index.js]: This is not right place to declare a function, index.js is entry point for Installer application. Windows related menu is defined in browsr/bootstrap.js file.

File: main/menu.js [hunk 3]
```diff
@@ -1,35 +1,12 @@
 'use strict';
```
Comment (dgolovin) [inline main/menu.js]: This file is also used on macOS to create macOS specific menu, this menu is not visible under Windows.

File: browser/about.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  <head>
+    <title>About</title>
+    <link rel="stylesheet" href="app.css">
+  </head>
+  <body>
+      <img class="image" style="margin-right:90px;margin-bottom:27px;" width="220px" src="images/RHDP_logo_RGB_default.svg"/>
+      <section class="welcome-panel">
+        <p>For help, please visit</p>
+        <ul>
+          <li><a href="https://access.redhat.com/documentation/en/red-hat-development-suite/">Documentation</a>.</li>
```
Comment (dgolovin) [inline browser/about.html]: These links loads referenced documents right in about window like this:

image

**Comment by dgolovin** [comment]
Here is PR https://github.com/redhat-developer-tooling/developer-platform-install/pull/841 that shows some of the issues fixed.

File: browser/pages/about/controller.js [hunk 5]
```diff
@@ -10,12 +10,16 @@ class AboutController {
...
-  next() {
-    this.router.go('location');
+  documentation() {
+    this.electron.shell.openExternal('https://access.redhat.com/documentation/en/red-hat-development-suite/');
   }
 
-  openDevSuiteOverview() {
-    this.electron.shell.openExternal('https://developers.redhat.com');
+  release() {
+    this.electron.shell.openExternal('https://access.redhat.com/documentation/en-us/red_hat_development_suite/1.4/html/release_notes_and_known_issues/');
```
Comment (dgolovin) [inline browser/pages/about/controller.js]: The installer version should not be hardcoded into the link. Use https://www.npmjs.com/package/semver module functions:
* major(version)
* minor(version)
to generate right url like
``` javascript
get shortVersion() {
  return `${semver.major($scope.version)}.${semver($scope.version).minor()}`;
}

release() {
  this.electron.shell.openExternal(`https://access.redhat.com/documentation/en-us/red_hat_development_suite/${this.shortVersion}/html/release_notes_and_known_issues/`);
}
```

File: browser/pages/about/controller.js [hunk 6]
```diff
@@ -10,12 +10,16 @@ class AboutController {
...
+  }
+
+  report() {
+    this.electron.shell.openExternal('https://access.redhat.com/documentation/en-us/red_hat_development_suite/1.4/html/installation_guide/troubleshooting#reporting_an_issue');
```
Comment (dgolovin) [inline browser/pages/about/controller.js]: use the same approach mentioned above.

File: browser/pages/about/about.html [hunk 7]
```diff
@@ -1,59 +1,14 @@
...
-  </div>
-
-</form>
+  <form>
+    <img class="image" style="margin-right:90px;margin-bottom:27px;" width="220px" src="images/logo_RHD_RGB_default.svg"/>
+    <span class="version-panel">Version: {{version}}</span>
+      <section class="welcome-panel">
+        <p>For help, please visit</p>
+        <ul>
+          <li><a href="" ng-click="aboutCtrl.documentation()">Documentation</a>.</li>
```
Comment (dgolovin) [inline browser/pages/about/about.html]: Dots are not required here.

File: browser/bootstrap.js [hunk 8]
```diff
@@ -39,18 +66,19 @@ const help = new MenuItem({
...
-  menu.popup(remote.getCurrentWindow());
-}, false);
-
 function restoreMenu() {
   menu = new Menu();
   menu.append(toggleDevToolsItem);
   menu.append(inspectElementItem);
   menu.append(separator);
   menu.append(help);
+  menu.append(aboutItem);
```
Comment (dgolovin) [inline browser/bootstrap.js]: ... and there should be separator between Help and About items.

**Comment by sudhirverma** [comment]
test this

**Comment by sudhirverma** [comment]
test this

File: browser/bootstrap.js [hunk 9]
```diff
@@ -27,6 +28,32 @@ const inspectElementItem = new MenuItem({
   }
 });
 
+function openAboutWindow() {
+
+  let aboutWindow = new BrowserWindow({
```
Comment (dgolovin) [inline browser/bootstrap.js]: Should be:
``` JavaScript
let aboutWindow = new BrowserWindow({
    parent: remote.getCurrentWindow(),
    modal: true,
    useContentSize: true,
    width: 565,
    height: 355,
    'autoHideMenuBar': true,
    resizable: false,
    show: false
  });
```
to keep original background size.

That would get us even closer to original design, but still vertical spacing between text requires some adjustments.

image

**Comment by sudhirverma** [comment]
aboutpage

**Comment by jrichter1** [comment]
Not a fan of this solution, especially when cmd+w shows greyed out in the main menu (still works though) until the modal window is closed. I would like it to be something within the app itself that I could click :)

Obviously, the nicest way would be to have some sort of close button as part of the dialog. We could also close the dialog when the main window regains focus (i.e. when the user clicks back into it), but like the current solution it might not be totally apparent thing to do.

Anyway, I'll let you merge this if you pledge to find a better way in this release.

**Comment by jrichter1** [comment]
test this

**Comment by sudhirverma** [comment]
test this

**Pull request merged**